src/components/S/ISelect.vue

```diff
@@ -39,8 +39,12 @@ const inputValue = computed({
   <SBase :definition="definition" :error="error" :dirty="dirty">
     <select v-model="inputValue" class="s-input">
       <option disabled value="">Please select one</option>
-      <option v-for="option in definition.enum" :key="option" :value="option">
-        {{ option }}
+      <option
+        v-for="option in definition.options || definition.enum"
+        :key="option"
+        :value="option?.id || option"
+      >
+        {{ option?.name || option }}
```

Comment:
Do we expect `option` to be undefined?

Reply:
No it's either a string or an object {id,name}

Reply:
We don't need `?.` then?

Reply:
yep